api/store/mongo/device_tags.go

```diff
@@ -35,19 +70,59 @@ func (s *Store) DevicePullTag(ctx context.Context, uid models.UID, tag string) e
 }
 
 func (s *Store) DeviceSetTags(ctx context.Context, uid models.UID, tags []string) (int64, int64, error) {
-	tag, err := s.db.Collection("devices").UpdateOne(ctx, bson.M{"uid": uid}, bson.M{"$set": bson.M{"tags": tags}})
+	tag, err := s.db.Collection("devices").
+		UpdateOne(ctx, bson.M{"uid": uid}, bson.M{"$set": bson.M{"tags": tags}})
```

Comment:
I have checked, this not use of the `MatchedCount` has not been made by myself, this could led for minimal changes in method behavior, they will remain consistent as to the rest of the other methods return value.